keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -459,87 +463,123 @@ Windows privileged containers will be implemented with [Job Objects](https://doc
 
-In windows, the same privileged field would need to be added in the windows pod security context (open issue) and the container security context will need to be augmented with the privileged flag.
+For Windows we are proposing the following updates to CRI-API:
 
-The new WindowsPodSandboxConfig:
-```
-message WindowsPodSandboxConfig {.
-	WindowsSandboxSecurityContext security_context = 1;
+Add WindowsPodSandboxConfig (and it to PodSandboxConfig)
+
+```protobuf
+message WindowsPodSandboxConfig {
+  WindowsSandboxSecurityContext security_context = 1;
 }
 ```
 
-A new WindowsSandboxSecurityContext:
-```
+Add WindowsSandboxSecurityContext:
+
+```protobuf
 message WindowsSandboxSecurityContext {
-	string run_as_user = 1;
-	bool privileged = 2;
+  string run_as_username = 1;
+  bool privileged = 2;
 }
 ```
 
-The new field for the WindowsContainerSecurityContext:
-```
+Update WindowsContainerSecurityContext by adding privilged field:
+
+```protobuf
 message WindowsContainerSecurityContext {
-	string run_as_username = 1;
-	string credential_spec = 2;
+  string run_as_username = 1;
+  string credential_spec = 2;
   bool privileged = 3;
 }
 ```
-### Kubelet Implementation Details
 
-#### <b>Privileged Flag</b>
-There are no kubernetes API changes required.  Kubelet can use the privileged flag from SecurityContext api and pass that to the new privileged flag in the CRI layer.  
+*Note:* For alpha annotaions on RunPodSandbox and CreateContainer CRI calls may be used untill a version of containerd with Windows privileged container support is released.
+
+### Kubernetes API updates
+
+#### Privilged Flag
+
+A new boolean field named `privileged` will be added to [WindowsSecurityContextOptions](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#windowssecuritycontextoptions-v1-core).
+
+On Windows, all containers in a pod mush be privileged. Because of this behavior and because `WindowsSecurityContextOptions` already exists on both `PodSecurityContext` and `Container.SecurityContext` Windows containers will use this new field instead of re-using the existing `privileged` field which only exists on `SecurityContext`.
 
-Add functionality to Kuberuntime_sandbox to split out the linux sandbox creation and add windows sandbox creation (WIP PR). 
+Having a new field will also help us better anticipate / validate user intention.
 
-https://github.com/kubernetes/kubernetes/blob/a9f1d72e1de6450b890a0c0e451725468f54f515/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L136 
+API validation will ensure that if `privileged` is set in `Container.SecurityContext.WindowsSercurityContextOptions` then `privileged` is also set in `PodSecurityContext.WindowsSecurityContextOptions`. Current behavior applies pod `WindowsSecurityContextOptions` to all container level `WindowsSecurityContextOptions` if not specified.
 
-#### <b>Host Network Mode</b>
+No additional validation will be performed against the existing `privileged` on `SecurityContext`.
```

Comment:
this seems problematic, see https://github.com/kubernetes/enhancements/pull/2288/files#r561978263

Reply:
Updated proposal to say we will be validating this field is set.